test/hotspot/gtest/utilities/test_vmerror.cpp

```diff
@@ -28,6 +28,17 @@
 
 #ifdef ASSERT
 
+#ifdef _WINDOWS
+
+#include <windows.h>
+
+TEST_VM_ASSERT_MSG(vmErrorTest, fatalWithError,
+                   "fatal error: GetLastError should be 6 - actually: 6") {
+  SetLastError(6);
+  fatal("GetLastError should be 6 - actually: %d", (int)GetLastError());
```

Comment:
I wonder if HotSpot has a preference for the more specific C\+\+ casts. Also, wouldn't it be better to check the value of GetLastError after fatal is called rather than comparing the strings (I assume that's what the TEST\_VM\_ASSERT\_MSG is doing)?

Reply:
@TheShermanTanker - This test is checking the problem use-case that led to this change, where the
value of the `GetLastError` call was clobbered when done as an argument to `fatal`, because the
`TOUCH_ASSERT_POISON` happened before the call to `GetLastError`.

Reply:
Ah, alright. Thanks for the explanation

Reply:
And `fatal` does not return. It ends the VM with an assertion message. google death tests scan for that message on stderr.